build.go

```diff
@@ -91,8 +92,14 @@ func Build(entryResolver EntryResolver,
 			launchMetadata.BOM = bom
 		}
 
-		cachedDependencySHA, ok := sdkLayer.Metadata["dependency-sha"]
-		if ok && cachedDependencySHA == sdkDependency.SHA256 { //nolint:staticcheck
+		dependencyChecksum := sdkDependency.Checksum
+		if sdkDependency.SHA256 != "" {
+			dependencyChecksum = sdkDependency.SHA256
+		}
+
+		cachedChecksum, ok := sdkLayer.Metadata["dependency-checksum"].(string)
```

Comment:
Now that I think about it, changing this key to `dependency-checksum` will break caching for users who upgrade, since the SDK layer will have a `dependency-sha`. Should we continue to suppport `dependency-sha`?

Reply:
I don't know that it is that steep of a penalty but I would be more than happy to make this a major release if that would help assuage worries.

Reply:
or we can leave the key named `dependency-sha`

Reply:
I do not have a strong feeling on this at all.